app/models/concerns/gobierto_common/has_external_id.rb

```diff
@@ -8,15 +8,27 @@ module HasExternalId
       before_validation :set_external_id
     end
 
+    module ClassMethods
+      def external_id_scope(method)
+        @external_id_method = method
+      end
+
+      def external_id_method
+        @external_id_method
+      end
+    end
+
     def set_external_id
       return if external_id.present?
 
-      self.external_id = new_external_id
+      self.external_id = self.class.external_id_method.present? ? scoped_new_external_id(send(self.class.external_id_method)) : new_external_id
```

Comment:
Why is necessary to set automatically the external_id if it's not present? Take into account that the external id can be a string, so the autoincrement you are proposing below won't work in all scenarios.

Reply:
We made the decision to set an id automatically if not present in plans projects to allow updates. This behavior was extracted to the concern the terms are using also. This method detects ids formed by digits, get the maximum and sets a new id with the maximum increased and converted to string, but it can coexist with arbitrary strings set by the user manually. I'll add an option to the concern to disable automatic id generation

Reply:
ok then!